Kentor.AuthServices/InvalidSAMLOperationException.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Kentor.AuthServices.Saml2P;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Kentor.AuthServices
+{
+    /// <summary>
+    /// Extended exception containing information about the status and status message SAML response.  
+    /// </summary>
+    [Serializable]
+    public class InvalidSamlOperationException: InvalidOperationException
```

Comment:
The exception types used in AuthServices is somewhat non-well-thought-through. This PR made me go back and see what I've done before and it's not very clear. But there is a bas class AuthServicesException that I think that this exception should derive from. What do you think?